Add Tenancy support to tunnels

p-rintz opened the pull request:
This PR adds support for tenancy to the tunnels.

I ran the tox tests, as well as extended the netbox tests with tenant support and ran those as well.

robertlynch3 commented:
Tenancy makes perfect sense.

Looks like the CI jobs are failing. 

```
IntegrityError: null value in column "tenant_id" of relation 
"netbox_tunnels2_tunnel" violates not-null constraint
DETAIL:  Failing row contains (3, 2023-07-21 16:53:31.528302+00, 2023-07-21 
16:53:31.528316+00, {}, Test Tunnel3, pending-config, null, null, , , , 1, 2, 
null, null, 1, null).


----------------------------------------------------------------------
Ran 2 tests in 1.228s

FAILED (errors=1)
Destroying test database for alias 'default'...
Error: Process completed with exit code 1.
```

I will see if I can dive into this.

p-rintz commented:
Hm, that looks like the error I got before I made tenancy optional in the model.
Which is strange, since it definitely is optional now.

Oh, I might have forgotten to copy the migrations, when they likely were changed afterwards.

I can look at that tomorrow if you haven’t done that yet.

p-rintz commented:
Couldnt help myself and looked at it just now.
It was the migrations.

robertlynch3 commented on netbox_tunnels2/migrations/0001_initial.py:
The changes need to be a new migration file, otherwise existing installs will fail.

I am working on the migration file now.

robertlynch3 commented:
I resolved the migration file issue in https://github.com/robertlynch3/netbox-tunnels2/commit/424bce3cde12c2d1c6054d4b812642395b7b48a3.

p-rintz commented:
I added the "new" selectors to forms on the A&B side of VRFs&Devices, and removed the tenant query param from the B side, as I dont think it fits very well there, by default.

With the selectors, it gets way less confusing to find a specific device/vrf.
This is especially true for the B side where the tenant may be different than the A side.
